source/blog/2017-02-16-hyperconverged-ovirt-with-ovn-part1.md

```diff
@@ -0,0 +1,203 @@
+```
+yum install -y cockpit-ovirt-dashboard gdeploy
+```
+On all the three hosts enable and start cockpit and add a firewall rule for it:
+```
+systemctl enable cockpit
+systemctl start cockpit
+firewall-cmd --zone=public --add-port=9090/tcp --permanent
+firewall-cmd --reload
+```
+If you are using VMs as your nodes, you should consider using a virtio-rng device on them to provide a source of entropy since going low on entropy can significantly slow down your demo env.
+```
+yum install -y epel-release
+yum install -y haveged
+systemctl start haveged
+systemctl enable haveged
+yum remove -y epel-release
+```
+If you are using EPEL please take care or removing it since it's going to make the installation of oVirt fail due to a dependency conflict.
+
+## Setup
+
+Now we can point your browser to https://demovm1.localdomain:9090/ to connect to cockpit running on your first host. Connect as root user.
+Point to Virtualization Tab and then Hosted Engine sub tab and choose there 'Hosted Engine with Gluster' as for picture 1.
+![2017-02-16-hyperconverged-ovirt-with-ovn-picture01.png](2017-02-16-hyperconverged-ovirt-with-ovn-picture01.png "2017-02-16-hyperconverged-ovirt-with-ovn-picture01.png")
+As the first step add the hostname of your three hosts on the *storage network*, since we have a single nic on each host I could simply use the FQDN but to make it more clear that's a different thing I’m going to use the host IP addresses as in picture 2.
+![2017-02-16-hyperconverged-ovirt-with-ovn-picture02.png](2017-02-16-hyperconverged-ovirt-with-ovn-picture02.png "2017-02-16-hyperconverged-ovirt-with-ovn-picture02.png")
+On the *Packages* step we can just keep everything as it is since we already installed what's needed.
+On the *Volumes* step it will propose by default to use three volumes but we are going to remove the vmstore one since we just want to build the minimal env for this demo lab so proceed as for picture 3. Please note that gdeploy will propose by default to use replica with arbiter mode to provide more space efficiency than a pure replica 3 mode.
+![2017-02-16-hyperconverged-ovirt-with-ovn-picture03.png](2017-02-16-hyperconverged-ovirt-with-ovn-picture03.png "2017-02-16-hyperconverged-ovirt-with-ovn-picture03.png")
+On the *Bricks* step remove the vmstore brick since we are not going to create the vmstore volume; point the device field to the storage device you are really going to use for the brick (vdb for both the volumes in my case); change the size of every brick to be sure they can fit on your device as in picture 4.
+![2017-02-16-hyperconverged-ovirt-with-ovn-picture04.png](2017-02-16-hyperconverged-ovirt-with-ovn-picture04.png "2017-02-16-hyperconverged-ovirt-with-ovn-picture04.png")
+Gdeploy will create the two gluster volumes and it will configure them to serve as best as possible for virtualization purposes. Wait until it successfully complete the volume definition as in picture 5.
```

Comment:
Here we are:

#gdeploy configuration generated by cockpit-gluster plugin
[hosts]
192.168.1.201
192.168.1.202
192.168.1.203

[disktype]
raid6

[diskcount]
12

[stripesize]
256

[yum1]
action=install
packages=vdsm-gluster,ovirt-hosted-engine-setup
update=yes
gpgcheck=yes
ignore_yum_errors=no

[shell1]
action=execute
command=sed -i 's/^\(server .*iburst\)/#\1/' /etc/ntp.conf

[update-file1]
action=add
dest=/etc/ntp.conf
line=server clock.redhat.com iburst

[service1]
action=enable
service=ntpd

[service2]
action=restart
service=ntpd

[shell2]
action=execute
command=vdsm-tool configure --force

[pv1]
action=create
devices=vdb
ignore_pv_errors=no

[vg1]
action=create
vgname=gluster_vg_vdb
pvname=vdb
ignore_vg_errors=no

[lv1]
action=create
poolname=gluster_thinpool_vdb
ignore_lv_errors=no
vgname=gluster_vg_vdb
lvtype=thinpool
poolmetadatasize=16GB
size=160GB

[lv2]
action=create
lvname=gluster_lv_engine
ignore_lv_errors=no
vgname=gluster_vg_vdb
mount=/gluster_bricks/engine
lvtype=thinlv
poolname=gluster_thinpool_vdb
virtualsize=80GB

[lv3]
action=create
lvname=gluster_lv_data
ignore_lv_errors=no
vgname=gluster_vg_vdb
mount=/gluster_bricks/data
lvtype=thinlv
poolname=gluster_thinpool_vdb
virtualsize=80GB

[selinux]
yes

[service3]
action=stop
service=NetworkManager

[service4]
action=disable
service=NetworkManager

[service5]
action=start
service=glusterd
slice_setup=yes

[firewalld]
action=add
ports=111/tcp,2049/tcp,54321/tcp,5900/tcp,5900-6923/tcp,5666/tcp,16514/tcp
services=glusterfs

[update-file2]
action=edit
dest=/etc/nagios/nrpe.cfg
replace=allowed_hosts
line=allowed_hosts=host.redhat.com

[service6]
action=restart
service=nrpe

[script1]
action=execute
file=/usr/share/ansible/gdeploy/scripts/disable-gluster-hooks.sh

[volume1]
action=create
volname=engine
transport=tcp
replica=yes
replica_count=3
key=group,storage.owner-uid,storage.owner-gid,features.shard,features.shard-block-size,performance.low-prio-threads,cluster.data-self-heal-algorithm,cluster.locking-scheme,cluster.shd-wait-qlength,cluster.shd-max-threads,network.ping-timeout,user.cifs,performance.strict-o-direct,network.remote-dio
value=virt,36,36,on,512MB,32,full,granular,10000,8,30,off,on,off
brick_dirs=/gluster_bricks/engine/engine
ignore_volume_errors=no
arbiter_count=1

[volume2]
action=create
volname=data
transport=tcp
replica=yes
replica_count=3
key=group,storage.owner-uid,storage.owner-gid,features.shard,features.shard-block-size,performance.low-prio-threads,cluster.data-self-heal-algorithm,cluster.locking-scheme,cluster.shd-wait-qlength,cluster.shd-max-threads,network.ping-timeout,user.cifs,performance.strict-o-direct,network.remote-dio
value=virt,36,36,on,512MB,32,full,granular,10000,8,30,off,on,off
brick_dirs=/gluster_bricks/data/data
ignore_volume_errors=no
arbiter_count=1

[yum2:192.168.1.201]
action=install
packages=ovirt-engine-appliance